client/galaxy/scripts/components/user/CustomBuilds.vue

```diff
@@ -0,0 +1,404 @@
+          <p class="card-text">For example, the first few entries of <em>mm9.len</em> are as follows:</p>
+          <pre class="card-text">chr1    197195432
+chr2    181748087
+chr3    159599783
+chr4    155630120
+chr5    152537259</pre>
+          <p class="card-text">Trackster uses this information to populate the select box for chrom/contig, andto set the maximum basepair of the track browser. You may either upload a .len fileof this format (Len File option), or directly enter the information into the box (Len Entry option).</p>
+        </b-card>
+      </b-col>
+    </b-row>
+  </b-container>
+</template>
+
+<script>
+import Vue from "vue";
+import BootstrapVue from "bootstrap-vue";
+import Multiselect from 'vue-multiselect';
+import axios from "axios";
+import Select from "../ui/Select.vue";
+
+Vue.use(BootstrapVue);
+
+export default {
+  components: {
+    "ui-select": Select,
+    Multiselect
+  },
+
+  data() {
+    return {
+      customBuildsUrl: `${Galaxy.root}api/users/${Galaxy.user.id}/custom_builds`,
+      installedBuilds: [],
+      maxFileSize: 100,
+      fileLoaded: 0,
+
+      fields: [
+        { key: "name", label: "Name" },
+        { key: "id", label: "Key" },
+        { key: "count", label: "Number of chroms/contigs" },
+        { key: "action", label: "" }
+      ],
+      items: [],
+
+      alertType: "info",
+      alertMessage: "",
+      dismissSecs: 5,
+      dismissCountDown: 0,
+
+      form: {
+        id: "",
+        name: "",
+        file: "",
+        text: ""
+      },
+
+      dataSources: [
+        { value: "fasta", text: "FASTA-file from history" },
+        { value: "file", text: "Len-file from disk" },
+        { value: "text", text: "Len-file by copy/paste" }
+      ],
+      selectedDataSource: null,
+      fastaFilesLoading: true,
+      fastaFilesSelectDisabled: true,
+      fastaFiles: [],
+      selectedFastaFile: null
+    };
+  },
+
+  computed: {
+    lengthType: function () {
+      return this.selectedDataSource ? this.selectedDataSource.value : "";
+    },
+
+    lengthValue: function () {
+      let value = "";
+
+      if (this.lengthType === "fasta") {
+        value = this.selectedFastaFile ? this.selectedFastaFile.value : "";
+      } else if (this.lengthType === "file") {
+        value = this.form.file;
+      } else if (this.lengthType === "text") {
+        value = this.form.text;
+      }
+
+      return value;
+    }
+  },
+
+  created() {
+    this.loadCustomBuilds();
+
+    Galaxy.currHistoryPanel.model.once("change", model => {
+      if (model.id) {
+        this.loadCustomBuildsMetadata(model.id);
+      } else {
+        this.fastaFilesLoading = false;
+      }
+    });
+  },
+
+  methods: {
+    loadCustomBuilds () {
+      axios
+        .get(this.customBuildsUrl)
+        .then(response => {
+          this.items = response.data;
+        })
+        .catch(error => {
+          console.log(error.response);
```

Comment:
These error messages should be shown in an alert message instead of the console.